### DISCUSSION TITLE
ImportError: cannot import name 'NO_KEY' from 'sqlalchemy.orm.base'

### DISCUSSION THREAD

**Discussion by C45513-L** [sqlalchemy/sqlalchemy#10455] [Usage Questions]
I got the error "from sqlalchemy.ext.automap import automap_base
from sqlalchemy.orm import Session" when I try to import the following class

from sqlalchemy.ext.automap import automap_base
from sqlalchemy.orm import Session

**Comment by CaselIT** [comment]
Hi,

That's strange. What version are you using?
These are usually venv issues, try re-installing it

**Comment by zzzeek** [comment]
no reason that should happen, make sure that you have SQLAlchemy freshly installed and it's not conflicting with any older version.

also please provide exact SQLAlchemy version in use as well as complete stack trace
